How to use .clamp() with a generic backend?

I fear I've probably missed something here, so a pointer in the right direction would be appreciated. As I understand it, we're supposed to define our model with just B: Backend, and only use a concrete definition in our entrypoint. We ran into trouble using .clamp() with this however - eg:

```rust
use burn::backend::WgpuBackend;
use burn::tensor::backend::Backend;
use burn::tensor::{Data, Tensor};

fn generic_code<B: Backend>() {
    let tensor = Tensor::<B, 2>::from_data(Data::<f32, 2>::from([[2., 3.], [4., 5.]]).convert());
    tensor.clamp(3., 4.);
}

fn main() {
    generic_code::<WgpuBackend>();
}
```

gives 

```
error[E0308]: arguments to this method are incorrect
   --> src/main.rs:7:12
    |
7   |     tensor.clamp(3., 4.);
    |            ^^^^^
    |
note: expected associated type, found floating-point number
```

I can work around this by changing the the function to 

```rust
fn generic_code<B: Backend<FloatElem = f32>() {
```

But that means we're no longer completely backend-agonstic. Is there a better way?

--- a/burn-import/src/burn/node/clip.rs
+++ b/burn-import/src/burn/node/clip.rs
@@ -1,5 +1,5 @@
 use super::{Node, NodeCodegen};
-use crate::burn::{BurnImports, Scope, TensorType, Type};
+use crate::burn::{Scope, TensorType, Type};
 use burn::record::PrecisionSettings;
 use proc_macro2::TokenStream;
 use quote::quote;
@@ -28,23 +28,20 @@ impl<PS: PrecisionSettings> NodeCodegen<PS> for ClipNode {
         if let Some(min) = self.min {
             if let Some(max) = self.max {
                 quote! {
-                    let #output = #input.clamp(#min.elem(), #max.elem());
+                    let #output = #input.clamp(#min, #max);
                 }
             } else {
                 quote! {
-                    let #output = #input.clamp_min(#min.elem());
+                    let #output = #input.clamp_min(#min);
                 }
             }
         } else if let Some(max) = self.max {
             return quote! {
-                let #output = #input.clamp_max(#max.elem());
+                let #output = #input.clamp_max(#max);
             };
         } else {
             panic!("Clip node must have at least one min or max value");
         }
-    }
-    fn register_imports(&self, imports: &mut BurnImports) {
-        imports.register("burn::tensor::ElementConversion");
     }
 
     fn into_node(self) -> Node<PS> {
@@ -73,7 +70,6 @@ mod tests {
         graph.register_input_output(vec!["tensor1".to_string()], vec!["tensor2".to_string()]);
 
         let expected = quote! {
-            use burn::tensor::ElementConversion;
             use burn::{
                 module::Module,
                 tensor::{backend::Backend, Tensor},
@@ -93,7 +89,7 @@ mod tests {
                 }
                 #[allow(clippy::let_and_return)]
                 pub fn forward(&self, tensor1: Tensor<B, 4>) -> Tensor<B, 4> {
-                    let tensor2 = tensor1.clamp(0f64.elem(), 1f64.elem());
+                    let tensor2 = tensor1.clamp(0f64, 1f64);
 
                     tensor2
                 }
@@ -117,7 +113,6 @@ mod tests {
         graph.register_input_output(vec!["tensor1".to_string()], vec!["tensor2".to_string()]);
 
         let expected = quote! {
-            use burn::tensor::ElementConversion;
             use burn::{
                 module::Module,
                 tensor::{backend::Backend, Tensor},
@@ -137,7 +132,7 @@ mod tests {
                 }
                 #[allow(clippy::let_and_return)]
                 pub fn forward(&self, tensor1: Tensor<B, 4>) -> Tensor<B, 4> {
-                    let tensor2 = tensor1.clamp_min(0f64.elem());
+                    let tensor2 = tensor1.clamp_min(0f64);
 
                     tensor2
                 }
@@ -161,7 +156,6 @@ mod tests {
         graph.register_input_output(vec!["tensor1".to_string()], vec!["tensor2".to_string()]);
 
         let expected = quote! {
-            use burn::tensor::ElementConversion;
             use burn::{
                 module::Module,
                 tensor::{backend::Backend, Tensor},
@@ -181,7 +175,7 @@ mod tests {
                 }
                 #[allow(clippy::let_and_return)]
                 pub fn forward(&self, tensor1: Tensor<B, 4>) -> Tensor<B, 4> {
-                    let tensor2 = tensor1.clamp_max(1f64.elem());
+                    let tensor2 = tensor1.clamp_max(1f64);
 
                     tensor2
                 }

--- a/burn-tensor/src/tensor/api/numeric.rs
+++ b/burn-tensor/src/tensor/api/numeric.rs
@@ -411,8 +411,8 @@ where
     /// # Returns
     ///
     /// A new tensor with the values clamped between the given min and max values.
-    pub fn clamp(self, min: K::Elem, max: K::Elem) -> Self {
-        Self::new(K::clamp(self.primitive, min, max))
+    pub fn clamp<E: ElementConversion>(self, min: E, max: E) -> Self {
+        Self::new(K::clamp(self.primitive, min.elem(), max.elem()))
     }
 
     /// Clamps a tensor under a minimum value.
@@ -425,8 +425,8 @@ where
     /// # Returns
     ///
     /// A new tensor with the values clamped under the given min value.
-    pub fn clamp_min(self, min: K::Elem) -> Self {
-        Self::new(K::clamp_min(self.primitive, min))
+    pub fn clamp_min<E: ElementConversion>(self, min: E) -> Self {
+        Self::new(K::clamp_min(self.primitive, min.elem()))
     }
 
     /// Clamps a tensor over a maximum value.
@@ -440,8 +440,8 @@ where
     ///
     /// A new tensor with the values clamped over the given max value.
     ///
-    pub fn clamp_max(self, max: K::Elem) -> Self {
-        Self::new(K::clamp_max(self.primitive, max))
+    pub fn clamp_max<E: ElementConversion>(self, max: E) -> Self {
+        Self::new(K::clamp_max(self.primitive, max.elem()))
     }
 
     /// Apply element wise absolute value operation

